src/lib/bootloader/grub2efi.rb

```diff
@@ -31,19 +32,28 @@ def read
 
     # Write bootloader settings to disk
     def write
-      # super have to called as first as grub install require some config writen in ancestor
+      # super have to called as first as grub install require some config written in ancestor
       super
 
       if pmbr_action && Yast::BootStorage.gpt_boot_disk?
-        efi_partition = Yast::Storage.GetEntryForMountpoint("/boot/efi")["device"]
-        efi_partition ||= Yast::Storage.GetEntryForMountpoint("/boot")["device"]
-        efi_partition ||= Yast::Storage.GetEntryForMountpoint("/")["device"]
-        efi_disk = Yast::Storage.GetDiskPartition(efi_partition)["disk"]
+        efi_partition = filesystems.find { |f| f.mountpoint == "/boot/efi" }
+        efi_partition ||= filesystems.find { |f| f.mountpoint == "/boot" }
+        efi_partition ||= filesystems.find { |f| f.mountpoint == "/" }
 
+        raise "could not find boot partiton" unless efi_partition
+
+        efi_partition = efi_partition.plain_blk_devices.first
+
+        efi_disk = efi_partition.disk
+
+# storage-ng
+# rubocop:disable Style/BlockComments
+=begin
         # get underlaying disk as it have to be set there and not on virtual one (bnc#981977)
         device = ::Bootloader::Stage1Device.new(efi_disk)
```

Comment:
I think this is not longer needed, as above is used "disk" method so it should be physical device, so can be removed.